src/main/java/eggventory/StockList.java

```diff
@@ -24,7 +21,50 @@ public StockList(ArrayList<StockType> stockList) {
     public StockList() {
         this.stockList = new ArrayList<>();
         this.stockList.add(new StockType("Uncategorised", false));
-        this.quantity = 0;
+    }
+
+    /**
+     * Gets the whole stockList. Note: technically doing using this method will violate OOP.
+     * @return the list.
+     */
+    public ArrayList<StockType> getList() {
+        return stockList;
+    }
+
+    /**
+     * Returns a stockType from stockList if it exits else retuns a null StockType.
+     * @param stockType The unique string that identifies a stockType
+     * @return stockType of stockList
+     */
+    public StockType getStockType(String stockType) {
+        StockType nullType = new StockType("NULL", true);
+        for (StockType stType : stockList) {
+            if (stType.getName().equals(stockType)) {
+                return stType;
+            }
+        }
+        return nullType;
+    }
+
+    /**
+     * Gets the total number of stockTypes in this stockList. Not to be confused with getStockQuantity.
+     * @return the number of stockTypes.
+     */
+    public int getStockTypeQuantity() { //The number of stockTypes in the list.
+        return stockList.size();
```

Comment:
I think quantity should be size - 1 though, because although uncategorised is a stock type, it likely wouldn't make sense to count the default dumping category.

Reply:
true. but by the end of my refactor we weren't using this method anymore. 

it used to be to iterate through the stocktypes, so counting uncategorised would be correct. however now we do "for stocktype in stocktypes" so it has become redundant. might remove in the future?